functionary/train_vision/train.py

```diff
@@ -0,0 +1,539 @@
+
+    model = LlavaLlamaForCausalLM.from_pretrained(
+        model_args.model_name_or_path,
+        torch_dtype=compute_dtype,
+        config=llava_cfg,
+        cache_dir=training_args.cache_dir,
+        use_flash_attention_2=True,
+    )
+
+    if hasattr(model.config, "tokenizer_model_max_length"):
+        model.config.tokenizer_model_max_length = training_args.model_max_length
+    model.config.use_cache = False
+
+    vision_tower = model.get_vision_tower()
+    if not vision_tower.is_loaded:
+        vision_tower.load_model(device_map="auto")
+    # if device_map != "auto":
+    #    vision_tower.to(device="cuda", dtype=torch.float16)
+    image_processor = vision_tower.image_processor
+
+    # model = LlavaLlamaForCausalLM.from_pretrained(model_path, low_cpu_mem_usage=True, attn_implementation=attn_implementation, config=llava_cfg, **kwargs)
+
+    print_rank0("Prompt template to use: ", training_args.prompt_template_version)
+    prompt_template = get_prompt_template_by_version(
+        training_args.prompt_template_version
+    )
+
+    tokenizer = initialize_tokenizer(
+        model=model,
+        model_name_or_path=model_args.model_name_or_path,
+        prompt_template=prompt_template,
+        model_max_length=training_args.model_max_length,
+        cache_dir=training_args.cache_dir,
+    )
+
+    if LOCAL_RANK == 0:
+        if not os.path.exists(training_args.output_dir):
+            os.mkdir(training_args.output_dir)
+
+        tokenizer_folder = os.path.join(training_args.output_dir, "tokenizer")
+        if not os.path.exists(tokenizer_folder):
+            os.mkdir(tokenizer_folder)
+        # Save tokenizer
+        tokenizer.save_pretrained(tokenizer_folder)
+
+    # get id of added tokens to compute the accuracy of predicing the token
+    id2token = {
+        tokenizer.encode(token)[-1]: token
+        for token in prompt_template.get_additional_tokens()
+    }
+    print_rank0("id to tokens: ", id2token)
+
+    assert data_args.train_data_path is not None, "Please provide a training data file."
+
+    with open(data_args.train_data_path, "r") as f:
+        raw_train_ds = [json.loads(line) for line in f]
+
+    train_dataset = LazyVisionDataset(raw_train_ds, tokenizer, data_args.pad_img_path)
+    if torch.distributed.get_rank() == 0:
+        print(f"Training Data Loaded: #{len(train_dataset)}")
+
+    if training_args.do_eval:
+        with open(data_args.eval_data_path, "r") as f:
+            raw_eval_ds = [json.loads(line) for line in f]
+
+        eval_dataset = LazyVisionDataset(raw_eval_ds, tokenizer, data_args.pad_img_path)
+
+        if torch.distributed.get_rank() == 0:
+            print(f"Eval Data Loaded: #{len(eval_dataset)}")
+
+    print_rank0("***** HERE ARE SOME EXAMPLES FROM TRAINING ****")
+    # print_some_examples(train_dataset, tokenizer)
+
+    print_rank0("***** HERE ARE SOME EXAMPLES FROM EVALUATION ***")
+    # print_some_examples(eval_dataset, tokenizer)
```

Comment:
We can remove this part if we dont print

Reply:
Sure